### PR TITLE
Web view sample Plugin for Theia

### PR DESCRIPTION
**Pull request by olexii4**
Add a sample plugin which includes:

- Creating and showing a basic webview.
- Dynamically updating a webview's content.
- Loading local content in a webview.
- Loading local tab-icon which depends on the current theme.
- Running scripts in a webview.
- Sending message from an extension to a webview.
- Sending messages from a webview to an extension.
- Using a basic content security policy.

demo

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

**Comment by AndrienkoAleksandr** [comment]
@olexii4 Did you used free icon, I mean with open source licence?

File: webview-sample-plugin/resources/main.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    window.addEventListener('message', function (event) {
+        const message = event.data; // The json data that the extension sent
+        switch (message.command) {
+            case 'refactor':
+                currentCount = Math.ceil(currentCount * 0.5);
+                counter.textContent = currentCount;
+                break;
+        }
+    })
+}());
```
Comment (AndrienkoAleksandr) [inline webview-sample-plugin/resources/main.js]: New line.

File: webview-sample-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "webview-sample",
+  "description": "Cat Coding - A Webview API Sample",
+  "publisher": "theia",
+  "keywords": [
+    "theia-plugin"
+  ],
+  "version": "0.0.1",
+  "license": "none",
```
Comment (AndrienkoAleksandr) [inline webview-sample-plugin/package.json]: Why licence is none?

File: webview-sample-plugin/resources/main.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                command: 'alert',
+                text: 'A typo at line ' + currentCount
+            });
+        }
+    }, 200);
+
+    // Handle messages sent from the extension to the webview
+    window.addEventListener('message', function (event) {
+        const message = event.data; // The json data that the extension sent
+        switch (message.command) {
```
Comment (AndrienkoAleksandr) [inline webview-sample-plugin/resources/main.js]: Do we really need switch for only one item? Maybe if would be better. What do You think?

File: webview-sample-plugin/resources/main.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// This script will be run within the webview itself
+// It cannot access the main VS Code APIs directly.
```
Comment (mmorhun) [inline webview-sample-plugin/resources/main.js]: Theia API?

File: webview-sample-plugin/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    output: {
+        filename: 'webview-sample-frontend.js',
+        
+        libraryTarget: "var",
+        library: "theia_webview_sample",
+                
+        path: path.resolve(__dirname, 'dist')
+    },
+    externals: {
+
```
Comment (mmorhun) [inline webview-sample-plugin/webpack.config.js]: please arrange config files

File: webview-sample-plugin/resources/main.js (hunk 1 above)
Comment (olexii4) [reply webview-sample-plugin/resources/main.js]: done

File: webview-sample-plugin/resources/main.js (hunk 4 above)
Comment (olexii4) [reply webview-sample-plugin/resources/main.js]: done. Thank you

File: webview-sample-plugin/webpack.config.js (hunk 5 above)
Comment (olexii4) [reply webview-sample-plugin/webpack.config.js]: It's from a plugin generator. Fixed.

File: webview-sample-plugin/resources/main.js (hunk 3 above)
Comment (olexii4) [reply webview-sample-plugin/resources/main.js]: It planned as several commands but totally agree.
Done

**Comment by olexii4** [comment]
@AndrienkoAleksandr  I added my own icons.

File: webview-sample-plugin/package.json (hunk 2 above)
Comment (olexii4) [reply webview-sample-plugin/package.json]: Good catch. It's from a plugin generator. Fixed.

**Comment by AndrienkoAleksandr** [comment]
@sunix I guess You can merge it

**Pull request merged**